thunder/executors/nvfuserex_impl.py

```diff
@@ -3016,6 +3016,72 @@ def cross_entropy_grad(
 )
 
 
+def _topk_check_(
+    a: TensorProxy, /, k: int, dim: int | None = None, largest: Number = 1, sorted: Number = 1, *args
+) -> bool:
+    if a.ndim <= 0:
+        return False
+    if dim >= a.ndim or (dim is not None and dim < -a.ndim):
+        return False
+    return True
+
+
+def topk_transform(
+    a: TensorProxy,
+    /,
+    k: int,
+    dim: int | None = None,
+    largest: Number = 1,
+    sorted: Number = 1,
+    *,
+    fd: FusionDefinition,
+    lc_to_nv_map: dict,
+) -> any:
+    nva = getnv(a, fd, lc_to_nv_map)
+    nvk = getnv(k, fd, lc_to_nv_map)
+    return fd.ops.topk(nva, nvk, dim, bool(largest), bool(sorted))
+
+
+register_supported(prims.topk, topk_transform, _topk_check_)
+
+
+def _argsort_check_(a: TensorProxy, /, dim: int | None = None, descending: bool = False, stable: bool = False) -> bool:
+    return True
+
+
+def argsort_transform(
+    a: TensorProxy,
+    /,
+    dim: int | None = None,
+    descending: bool = False,
+    stable: bool = False,
+    *,
+    fd: FusionDefinition,
+    lc_to_nv_map: dict,
+) -> TensorProxy:
+    """Transform argsort operation for NVFuser execution.
+
+    Args:
+        a: Input tensor
+        dim: Dimension along which to sort
+        descending: Sort in descending order if True
+        stable: Whether to use a stable sorting algorithm
+        fd: Fusion definition
+        lc_to_nv_map: Map from Lightning tensors to NVFuser tensors
+
+    Returns:
+        TensorProxy: Tensor of indices that would sort the input tensor
+    """
+    nva = getnv(a, fd, lc_to_nv_map)
+    if dim is None:
+        dim = a.ndim - 1 if a.ndim > 0 else 0
+
+    return fd.ops.argsort(nva, dim, bool(descending), bool(stable))
```

Comment:
Is this an actual primitive in nvFuser or it's decomposed into some other primitives internally?